### PR TITLE
AMPI: Improve user-friendliness of toolchain wrappers

### PR DESCRIPTION
**Pull request by evan-charmworks**
* Add "ampic++" as synonym for ampiCC
* Add "ampifort" as synonym for ampif90
* Add "ampiexec" as synonym for ampirun
* Add mpi*.ampi variants of all ampi* scripts to match MPICH and OpenMPI
* Add bin/ampi/ subfolder with standard mpi* names for PATH purposes

### CODE REVIEW

**Comment by ericjbohm** [comment]
Is there still an assumption that mpiexec interfaces with mpd?

**Comment by evan-charmworks** [comment]
> Is there still an assumption that mpiexec interfaces with mpd?

I'm going by the behavior of OpenMPI and MPICH:

```
$ realpath `which mpirun.openmpi`
/usr/bin/orterun
$ realpath `which mpiexec.openmpi`
/usr/bin/orterun
$ realpath `which mpirun.mpich`
/usr/bin/mpiexec.hydra
$ realpath `which mpiexec.mpich`
/usr/bin/mpiexec.hydra
```

File: src/libs/ck-libs/ampi/CMakeLists.txt [hunk 1]
```diff
@@ -18,14 +18,38 @@ if(${CMK_CAN_LINK_FORTRAN})
...
+configure_file(ampiCC        ${CMAKE_BINARY_DIR}/bin/mpicc.ampi     COPYONLY)
+configure_file(ampiCC        ${CMAKE_BINARY_DIR}/bin/mpiCC.ampi     COPYONLY)
+configure_file(ampiCC        ${CMAKE_BINARY_DIR}/bin/mpicxx.ampi    COPYONLY)
+configure_file(ampiCC        ${CMAKE_BINARY_DIR}/bin/mpic++.ampi    COPYONLY)
+configure_file(ampif77       ${CMAKE_BINARY_DIR}/bin/mpif77.ampi    COPYONLY)
+configure_file(ampif90       ${CMAKE_BINARY_DIR}/bin/mpif90.ampi    COPYONLY)
+configure_file(ampif90       ${CMAKE_BINARY_DIR}/bin/mpifort.ampi   COPYONLY)
+configure_file(ampirun       ${CMAKE_BINARY_DIR}/bin/mpirun.ampi    COPYONLY)
+configure_file(ampirun       ${CMAKE_BINARY_DIR}/bin/mpiexec.ampi   COPYONLY)
+make_directory(${CMAKE_BINARY_DIR}/bin/ampi/)
```
Comment (matthiasdiener) [inline src/libs/ck-libs/ampi/CMakeLists.txt]: Should be `file(MAKE_DIRECTORY ${CMAKE_BINARY_DIR}/bin/ampi/)`

File: src/libs/ck-libs/ampi/pathstub.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+DIR=$(dirname $0)
+FN=$(basename $0)
+
+# detect and handle circular calls when:
+# * building with the MPI machine layer
+# * using Charmrun's ++mpiexec
+if [[ "$FROM_CHARMC" = '1' ]] || [[ "$FROM_CHARMRUN" = '1' ]]; then
```
Comment (matthiasdiener) [inline src/libs/ck-libs/ampi/pathstub.sh]: could be `[[ "$FROM_CHARMC" = '1' || "$FROM_CHARMRUN" = '1' ]]` I believe.

File: src/arch/mpi-bluegeneq/charmrun [hunk 3]
```diff
@@ -3,6 +3,8 @@
 # good for job less than 30 minutes
 # have to pass bank using -A or set PROJECT environment variable
 
+export FROM_CHARMRUN='1'
```
Comment (matthiasdiener) [inline src/arch/mpi-bluegeneq/charmrun]: What's the purpose of these `FORM_CHARMRUN`, `FROM_CHARMC` variables?

Comment (evan-charmworks) [reply src/arch/mpi-bluegeneq/charmrun]: See the if in `pathstub.sh`

**Comment by evan-charmworks** [comment]
Ready for review

**Pull request merged**